oeps/architectural-decisions/oep-0064-arch-mobile-application-codebase-modernization.rst

```diff
@@ -0,0 +1,99 @@
+
+* Support for the existing mobile applications will continue through the Redwood release. Starting with the Quince release, we will designate the new repositories as the default Open edX mobile applications to encourage adoption and use of the new applications for community team members.
+* Once available, we will link to documentation within the new repositories to describe transition options for community members migrating from existing builds of the older codebases.
+* After the publication of the Quince release we will move the original repositories or the openedx-unsupported github organization to help make clear that these are no longer the default mobile application builds moving forward. A corresponding DEPR ticket will be linked to this OEP once it exists. 
+
+The new mobile repositories are maintained by Raccoon Gang and can be accessed here:
+* https://github.com/raccoongang/educationx-app-ios
+* https://github.com/raccoongang/educationx-app-android
+
+* The new repositories will be migrated to the ``openedx`` github organization under the names ``openedx-app-ios`` and ``openedx-app-android``.
+* The new repositories will be maintained in accordance with OEP-55.
+* Relevant funded contributions and projects relation to the new mobile repositories will be linked from their main project README as a way to help community members find ongoing and future efforts.
+
+
+Rationale
+*********
+
+Expanded access and opportunities for learners around the globe means meeting learners where they are, and for many people that means on their only digital device, their phone. Furthermore, enabling learners to continue to learn on their desktop or laptop computers in addition to a full featured mobile experience allows us to support full day learning workflows for those with multiple devices.
+
+Similar to the way the edx-platform has undergone a modernization process over the past few years with updated build processes, micro services, micro frontends, design system tools and more, the mobile experiences will benefit from a modern architecture, being written in the latest development language tools, and set up for increased pluggability and contribution as well.
+
+
+Reference Implementation
+************************
+
+The new mobile repositories are maintained by Raccoon Gang and can be accessed here:
+
+* https://github.com/raccoongang/educationx-app-ios
+* https://github.com/raccoongang/educationx-app-android
+
+Rejected Alternatives
+*********************
+
+We considered exploring the development costs of incrementally improving the existing mobile applications to fully adopt new architectural patterns as well as the use of Swift and Kotlin. Efforts to use these new technologies go back many years, but it would take considerable effort for the community to support this migration. Realistically the current team at edX / 2U maintaining the repositories would have had to undertake a major development effort to achieve this transformation. 
```

Comment:
[curious/clarification] What other technologies/architectural patterns were considered beyond Swift and Kotlin?

For example, was [React Native](https://reactnative.dev/) explored? While I personally don't have much experience with React Native, I believe one of its primary goals is to enable a "write one, use everywhere" approach where possible, enabling sharing of code across platforms (even between web + mobile). One might hypothesize having shared code across iOS, Android, and web could lead to having the mobile apps be more approachable/maintainable by the community, align closer to other React-based UI development in the Open edX platform (i.e., micro-frontends), etc.

More specifically, I'd be curious whether something like React Native could enable sharing of existing styles/patterns/functionality in the Paragon design system. For example, could common UI patterns across both web and mobile both live under the same Paragon design system, but ship with and get consumed via different implementations (React Native for mobile vs. React for web)? Food for thought; felt appropriate to expand a bit more on the technology choices.